### PR TITLE
LCH-6268: Fixing phpstan issues.

### PR DESCRIPTION
**Pull request by narendradesai**
Issue: https://backlog.acquia.com/browse/LCH-6268

### CODE REVIEW

File: phpstan.neon [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    - '#PHPDoc tag @var for variable [a-zA-Z0-9_\$]* contains unknown class [a-zA-Z0-9\\_:@\(\).]*#'
+    - '#PHPDoc tag @throws with type [a-zA-Z0-9\\_:@\(\)|]* is not subtype of Throwable#'
+    - '#Class [a-zA-Z0-9\\_:@\(\)]* not found.#'
+    - '#Parameter [a-zA-Z0-9_\$]* of method [a-zA-Z0-9\\_:@\(\)]* has invalid type [a-zA-Z0-9\\_:@\(\).]*#'
+    - '#Method [a-zA-Z0-9\\_:@\(\)]* has invalid return type [a-zA-Z0-9\\_:@\(\).]*#'
+    - '#Instantiated class [a-zA-Z0-9\\_:@\(\).]*#'
+    - '#Unsafe usage of new static\(\).#'
+    - '#Access to constant [a-zA-Z0-9_\$]* on an unknown class [a-zA-Z0-9\\_:@\(\).]*#'
+    - '#Caught class [a-zA-Z0-9\\_:@\(\)]* not found.#'
+    - '#Throwing object of an unknown class [a-zA-Z0-9\\_:@\(\).]*#'
```
Comment (saxumVermes) [inline phpstan.neon:16]: That's a considerable amount of ignores. Could you please leave an explanation for each point?

Comment (saxumVermes) [reply phpstan.neon:16]: Do we need a major refactor?

Comment (narendradesai) [reply phpstan.neon:16]: @saxumVermes  These error-ignores are for the classes/methods which are not available within the scope of package. Eg. `Call to static method service() on an unknown class Drupal.`

Comment (narendradesai) [reply phpstan.neon:16]: I will leave an error example for each point.

File: phpstan.neon [hunk 2]
```diff
@@ -0,0 +1,18 @@
+parameters:
+  level: 3
+  paths:
+    - ./src
+  ignoreErrors:
+    - '#Call to [a-zA-Z0-9\\_:@\(\).]*#'
```
Comment (narendradesai) [inline phpstan.neon:6]: E.g:

1. `Call to static method service() on an unknown class Drupal.`
2. `Call to static method getContainer() on an unknown class Drupal.`
3. `Call to an undefined method Acquia\Console\ContentHub\Tests\Drupal\Command\ContentHubAuditCheckUuidTest::assertNotContains().`

File: phpstan.neon [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+    - '#PHPDoc tag @var for variable [a-zA-Z0-9_\$]* contains unknown class [a-zA-Z0-9\\_:@\(\).]*#'
```
Comment (narendradesai) [inline phpstan.neon:7]: E.g.

- `PHPDoc tag @var for variable $client_manager contains unknown class \Drupal\acquia_contenthub\Client\ClientManagerInterface.`

File: phpstan.neon [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+    - '#PHPDoc tag @throws with type [a-zA-Z0-9\\_:@\(\)|]* is not subtype of Throwable#'
```
Comment (narendradesai) [inline phpstan.neon:8]: E.g: `PHPDoc tag @throws with type \Drupal\Component\Plugin\Exception\InvalidPluginDefinitionException is not subtype of Throwable`

File: phpstan.neon [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+    - '#Class [a-zA-Z0-9\\_:@\(\)]* not found.#'
```
Comment (narendradesai) [inline phpstan.neon:9]: E.g: `Class Drupal\Core\Config\Entity\ConfigEntityType not found.`

File: phpstan.neon [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+    - '#Parameter [a-zA-Z0-9_\$]* of method [a-zA-Z0-9\\_:@\(\)]* has invalid type [a-zA-Z0-9\\_:@\(\).]*#'
```
Comment (narendradesai) [inline phpstan.neon:10]: E.g: `Parameter $config of method Acquia\Console\ContentHub\Command\ContentHubAuditChSettings::syncSettings() has invalid type Drupal\Core\Config\Config.`

File: phpstan.neon [hunk 7]
```diff
@@ -0,0 +1,18 @@
...
+    - '#Method [a-zA-Z0-9\\_:@\(\)]* has invalid return type [a-zA-Z0-9\\_:@\(\).]*#'
```
Comment (narendradesai) [inline phpstan.neon:11]: E.g: `Method getConfigEntitiesWithMissingUuid() has invalid return \Drupal\Core\Config\Config.`

File: phpstan.neon [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+    - '#Instantiated class [a-zA-Z0-9\\_:@\(\).]*#'
```
Comment (narendradesai) [inline phpstan.neon:12]: E.g: `Instantiated class Drupal\Core\Extension\MissingDependencyException not found.`

File: phpstan.neon [hunk 9]
```diff
@@ -0,0 +1,18 @@
...
+    - '#Unsafe usage of new static\(\).#'
+    - '#Access to constant [a-zA-Z0-9_\$]* on an unknown class [a-zA-Z0-9\\_:@\(\).]*#'
```
Comment (narendradesai) [inline phpstan.neon:14]: E.g: `Access to constant MAXIMUM_NONE on an unknown class Drupal\Core\Cache\DatabaseBackend.`

File: phpstan.neon [hunk 10]
```diff
@@ -0,0 +1,18 @@
...
+    - '#Caught class [a-zA-Z0-9\\_:@\(\)]* not found.#'
```
Comment (narendradesai) [inline phpstan.neon:15]: E.g: `Caught class Drupal\Component\Plugin\Exception\PluginNotFoundException not found.`

**Comment by narendradesai** [comment]
@saxumVermes I've added example for  each error ignore, these are mostly because classes are available outside the scope of package.

**Review by saxumVermes — APPROVED** [review]
Looks good, thanks for the explanations!

**Pull request merged**